packages/main/src/plugin/container-registry.ts

```diff
@@ -554,6 +554,46 @@ export class ContainerProviderRegistry {
     return flattenedContainers;
   }
 
+  // Assuming ImageInfo, ListImagesOptions, and PodmanListImagesOptions types are defined elsewhere
+  async unifiedListImages(options?: ListImagesOptions & PodmanListImagesOptions): Promise<ImageInfo[]> {
+    let telemetryOptions = {};
+
+    // This gets all the available providers if no provider has been specified
+    let providers: InternalContainerProvider[];
+    if (options?.provider === undefined) {
+      providers = Array.from(this.internalProviders.values());
+    } else {
+      providers = [this.getMatchingContainerProvider(options.provider)];
+    }
+
+    const images = await Promise.all(
+      providers.map(async provider => {
+        try {
+          if (provider.libpodApi) {
+            console.log(`Using libpodApi for provider ${provider.name}`);
+            return await this.podmanListImages({ provider: provider.connection });
```

Comment:
it looks strange, there is no need to convert fields between the two format ? libpod API and compat API ?